diagnostic_feedback/quiz.py

```diff
@@ -241,6 +242,35 @@ def student_view(self, context=None):
             }
         )
 
+    def student_view_data(self, context=None):
+        """
+        Returns a JSON representation of the Diagnostic Feedback Xblock, that
+        can be retrieved using Course Block API.
+        """
+        return {
+            'quiz_type': self.quiz_type,
+            'quiz_title': self.title,
+            'questions': self.questions,
+        }
+
+    @XBlock.handler
+    def student_view_user_state(self, data, suffix=''):
+        """
+        Returns a JSON representation of the student data for Diagnostic Feedback Xblock
+        """
+        response = {
+            'student_choices': self.student_choices,
+            'student_result': self.student_result,
```

Comment:
It looks like `self.student_result` contains only the category name, but no other info such as category html_body or image. We might want to replace `self.student_result` with `self.get_result()`, although that's also not ideal as it returns HTML.

I don't know enough about this XBlock to know what the best approach would be, so I think `self.student_result` is fine for the first version, but we might want to expand it in the future.